packages/nextjs/components/ExampleUi/ContractInteraction/PurposeSetter.tsx

```diff
@@ -1,6 +1,23 @@
+import { useState } from "react";
 import { ArrowSmallRightIcon } from "@heroicons/react/24/outline";
+import { useScaffoldContractWrite, useTransactor } from "~~/hooks/scaffold-eth";
+import { getParsedEthersError } from "~~/components/scaffold-eth/Contract/utilsContract";
+import { toast } from "~~/utils/scaffold-eth";
 
 export const PurposeSetter = () => {
+  const [newPurpose, setNewPurpose] = useState("");
+
+  // Could we move useTransactor inside useScaffoldContractWrite?
+  const writeTxn = useTransactor();
```

Comment:
Could we move useTransactor inside useScaffoldContractWrite?

Reply:
Haven't thought about the limitations of doing so, but doing it will also remove the need of doing `if (writeAsync && writeTxn) {` when calling the write TX

Reply:
`if (writeAsync && writeTxn) { ` yeah this is a bit annoying and we definitely need to abstract it, also I see many people many making mistake here `await writeTxn(writeAsync())` they might forget to call `writeAsync` and it's better to handle it in `useScaffolContractWrite`

### Suggestion : 
Maybe we can use `useTransactor` inside `useScaffoldContractWrite` and wrap wagmi's `writeAsync` and then pass it to the user of the hook? 

Umm regarding inlining `useTransactor` in `useScaffoldContractWrite`, like `useTransactor` can also be handled to send raw txn's like in Faucet.tsx https://github.com/scaffold-eth/se-2/blob/0bf3a09a3b47017ad18e5ee4cf79ce83f4fc9628/packages/nextjs/components/scaffold-eth/Faucet.tsx#L29

Like I really feel we need to handle this `if (writeAsync && writeTxn) { ` internally and we can may be use the `suggestion` mentioned above instead of putting in `useTransactor` in `useScaffoldContractWrite` but happy to take either approach 🙌

Reply:
> Umm regarding inlining useTransactor in useScaffoldContractWrite, like useTransactor can also be handled to send raw txn's like in Faucet.tsx

I think it's better to use `useTransactor` and that `if` inside `useScaffoldContractWrite`, but also use `useTransactor` for raw transactions. Or maybe create a different hook `useRawTransactor` for raws

Reply:
> Or maybe create a different hook `useRawTransactor` for raws
Even this is a great idea! if we go with `useRawTransactor` then we can eliminate `useTransactor` 🙌

Reply:
Great ideas!

I tried this: https://github.com/scaffold-eth/se-2/pull/124/commits/e09915c72b43951c318629619b47c1d2ca130f3a

It was the shortest path possible. I love the `onClick={writeAsync}` use for the hook user :D

What do you all think?